core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/RepositoryTransactionContext.java

```diff
@@ -277,8 +277,8 @@ public Long commit(String author, String commitComment, String parentLockContext
 		final DatastoreLockTarget lockTarget = createLockTarget(id(), path());
 		IOperationLockManager locks = service(IOperationLockManager.class);
 		Commit commit = null;
+		acquireLock(locks, lockContext, lockTarget);
```

Comment:
I don't remember how the locking goes internally, but if some locks can be granted when another fails, the granted ones should be released. In this case, `acquireLock` should remain inside the `try-finally` block.

Reply:
Hmm, right, then we need to omit/skip locks that cannot be released. 😕